pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -932,6 +963,20 @@ func (r *ReconcileClusterDeployment) addClusterDeploymentFinalizer(cd *hivev1.Cl
 }
 
 func (r *ReconcileClusterDeployment) removeClusterDeploymentFinalizer(cd *hivev1.ClusterDeployment) error {
+
+	if cd.Spec.ManageDNS {
```

Comment:
Not sure that this is needed. All calls to removeClusterDeploymentFinalizer occur in the same function that calls `ensureManagedDNSZoneDeleted` above. And because we're using foreground propagation, the DNSZone should be gone by this point.

Reply:
The if or the the whole block? 

This one relates to the namespace deletion, which UHC assumes it can do as soon as the cluster deployment is gone. We saw this in stage where the cluster deployment is gone, the namespace is deleted, the dnszone has a deletionTimestamp after the ns was deleted, but can't act because the AWS creds secret also was deleted. This part of the fix makes sure the cluster doesn't go away until the DNSZone does (which Kube should be doing for us anyhow, but doesn't seem to 100% reliably), which in turn keeps UHC from deleting the namespace until we're ready. (alternative would be asking them to wait for both cluster deployment and dnszone to be gone which they're willing to do, but seemed best to handle this internally)

Reply:
Or a rephrase, the other part of this fix ensures the DNSZone gets a delete request, this part ensures we wait for it to complete before we let the CD go away which would cause UHC to delete the namespace.

Reply:
The whole block :) 
I didn't see another path to this block except after calling the `ensureManagedDNSZoneDeleted` function you introduced above. And if you've already called that function, then this seems unnecessary.

https://github.com/openshift/hive/blob/9fb4db000c0108175785f3a09230960ca990a49a/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L856-L957

Reply:
my understanding of DeletePropagationForeground is that the delete will wait for the resource to actually go away

Reply:
I see, that makes sense, will remove.

Reply:
done